src/devnet/integrated-devnet-logger.ts

```diff
@@ -0,0 +1,76 @@
+import * as path from "path";
+import * as fs from "fs";
+
+export class IntegratedDevnetLogger {
+
+    // Promisify fs
+    protected fsPromises = fs.promises;
+
+    constructor(protected stdout?: string, protected stderr?: string) {
+        if (this.logToFile(this.stdout)) {
+            this.checkFileExists(this.stdout);
+        }
+
+        if (this.logToFile(this.stderr)) {
+            this.checkFileExists(this.stderr);
+        }
+    }
+
+    // Checks if the file exists
+    private async checkFileExists(filePath: string): Promise<void> {
+        const file = path.resolve(filePath);
+        // Create the file if it doesn't exist
+        const dir = path.dirname(file);
+        if (!fs.existsSync(dir)) {
+            await this.fsPromises.mkdir(dir, { recursive: true });
+        }
+
+        if (!fs.existsSync(file)) {
+            await this.fsPromises.writeFile(file, "");
+        }
+    }
+
+    public async logStdout(message: string): Promise<void> {
+        // STDOUT
+        if (this.stdout === "STDOUT") {
```

Comment:
stdout shouldn't be able to just log to STDOUT, it should be able to log to e.g. STDERR. You might think of abstracting these loggers to some classes. Also, how is the case of this.stdout being undefined handled?